uaclient/api/u/pro/auto_attach/with_short_retry/v1.py

```diff
@@ -0,0 +1,243 @@
+                messages.UNSUPPORTED_AUTO_ATTACH
+            )
+        raise e
+
+    token = tokenResponse["contractToken"]
+    return token
+
+
+def with_short_retry(options: AutoAttachWithShortRetryOptions, cfg: UAConfig):
+    i = 0
+    already_attached = False  # avoid AlreadyAttachedOnProError after 1st time
+    while i <= 2:
+        if not already_attached:
+            instance = None  # type: Optional[AutoAttachCloudInstance]
+            token = None  # Optional[str]
+            try:
+                instance = get_cloud_instance(cfg)
+            except exceptions.UserFacingError as e:
+                raise CloudError(
+                    msg=str(e.msg), msg_code="unable-to-complete-cloud-checks"
+                )
+            if not instance:
+                raise AutoAttachDisabledError(
+                    msg="UA Config disable_auto_attach is set."
+                )
+            try:
+                detach_before_auto_attach(cfg)
+            except exceptions.AlreadyAttachedOnPROError as e:
+                raise ProAlreadyAttachedError(
+                    msg=str(e.msg), msg_code="pro-already-attached"
+                )
+            except exceptions.UserFacingError as e:
+                raise DetachError(msg=str(e.msg), msg_code="unable-to-detach")
+            allow_enable = not any([options.enable, options.enable_beta])
+            try:
+                token = get_auto_attach_token(cfg, instance)
+                actions.attach_with_token(
+                    cfg, token=token, allow_enable=allow_enable
+                )
+                if allow_enable:
+                    return AutoAttachWithShortRetryResult(successful=True)
+            except exceptions.UrlError as e:
+                raise AutoAttachError(
+                    msg=str(e.strerror), msg_code="unable-to-get-token"
+                )
+            except exceptions.UserFacingError as e:
+                raise AutoAttachError(
+                    msg=str(e.msg), msg_code="unable-to-get-token"
+                )
+            already_attached = True
+
+        services = list(set(options.enable + options.enable_beta))
+        found, not_found = cli.get_valid_entitlement_names(services, cfg)
+        enabled_services = 0
+
+        for name in found:
+            ent_ret, reason = actions.enable_entitlement_by_name(
+                cfg, name, assume_yes=True, allow_beta=True
+            )
+            if not ent_ret:
+                if (
+                    reason is not None
+                    and isinstance(reason, CanEnableFailure)
+                    and reason.message is not None
+                ):
+                    event.info(reason.message.msg)
+                    event.error(
+                        error_msg=reason.message.msg,
+                        error_code=reason.message.name,
+                        service=name,
+                    )
+                    # Should we raise an Error?
```

Comment:
We should not have `event` calls in the api module. Everything that goes wrong here need to be raised using an Exception